neqo-http3/src/recv_message.rs

```diff
@@ -167,12 +167,8 @@ impl RecvMessage {
         }
 
         let is_web_transport = self.message_type == MessageType::Request
-            && headers
-                .iter()
-                .any(|h| h.name() == ":method" && h.value() == "CONNECT")
-            && headers
-                .iter()
-                .any(|h| h.name() == ":protocol" && h.value() == "webtransport");
+            && headers.contains(&Header::new(":method", "CONNECT"))
```

Comment:
`Header::new` allocates both key and value into a `String`. Is that an issue here?

https://github.com/mozilla/neqo/blob/a7581770ff138c16727e92ecb7b47d9ceb95425e/neqo-common/src/header.rs#L13-L23

If yes, we could have `Header` wrap `Cow<'static, str>`, or introduce a `BorrowedHeader`.

Reply:
Either works and might save a tiny amount of work.  Though we are avoiding most of the work on the client side here, so it's not that important.

Reply:
Reflecting on this, perhaps we could implement `PartialEq<(&str, &str)> for Header` and avoid all of that complication.  Then we'd only have to delete the string `Header::new` from these...

Reply:
https://github.com/larseggert/neqo/pull/32 is where I got.  It turns out that `slice::contains` doesn't have the right signature.